### PR TITLE
PA: 212, PA: 1155 - Get Data service PVC to full condition and resize

### PR DESCRIPTION
**Pull request by jkagliwal-px**
This PR is with respect to two JIRA tickets -

https://portworx.atlassian.net/browse/PA-1155 (Create a function to poll for PVC usage and resize after full usage)
https://portworx.atlassian.net/browse/PA-212 (testcase - Get Data service PVC to full condition and resize the pvc)
Enhancements made ->

Used portworx libraries to get the volumes and inspect the usage of the pvc.
After reaching a threshold, resizing the pvc by 1 gb.
To fill the pvc, pds workload scripts are executed.
Introduced a new function to pass custom resource templates from the testcase.
PS: 1.This testcase currently runs only for Postgres as I need to pass a custom resource template with a small size storage for faster test execution.

Jenkins Run -

https://jenkins.pwx.dev.purestorage.com/job/PDS/job/pds-branch-build-test/375/console

### CODE REVIEW

File: tests/pds/dataservice_test.go [hunk 1]
```diff
@@ -1666,3 +1667,97 @@ var _ = Describe("{AddAndValidateUserRoles}", func() {
 		EndTorpedoTest()
 	})
 })
+
+var _ = Describe("{GetPvcToFullCondition}", func() {
+
+	JustBeforeEach(func() {
+		InitInstance()
```
Comment (Madanagopal19) [inline tests/pds/dataservice_test.go:1674]: Add InitInstance in BeforeSuite

File: tests/pds/dataservice_test.go [hunk 2]
```diff
@@ -1666,3 +1667,97 @@ var _ = Describe("{AddAndValidateUserRoles}", func() {
...
+	It("Deploy Dataservices", func() {
+		var generateWorkloads = make(map[string]string)
+		var deployments = make(map[PDSDataService]*pds.ModelsDeployment)
+		var dsVersions = make(map[string]map[string][]string)
+		var depList []*pds.ModelsDeployment
+		var dsName string
+
+		Step("Deploy Data Services", func() {
+			for _, ds := range params.DataServiceToTest {
+				Step("Deploy and validate data service", func() {
```
Comment (Madanagopal19) [inline tests/pds/dataservice_test.go:1687]: If this test is only for postgres, use if condition here as well to run only for pg data service

File: tests/pds/common.go [hunk 3]
```diff
@@ -1,16 +1,21 @@
 package tests
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/portworx/torpedo/drivers/pds/parameters"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/tests"
```
Comment (Madanagopal19) [inline tests/pds/common.go:9]: why do you need to import tests in same package?

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]

File: tests/pds/common.go [hunk 4]
```diff
@@ -162,3 +167,81 @@ func RunWorkloads(params pdslib.WorkloadGenerationParams, ds PDSDataService, dep
 	return pod, dep, err
 
 }
+
+// Check the DS related PV usage and resize in case of 90% full
+func CheckPVCtoFullCondition(context []*scheduler.Context) error {
+	log.Infof("Check PVC Usage")
+	f := func() (interface{}, bool, error) {
+		for _, ctx := range context {
+			vols, err := tests.Inst().S.GetVolumes(ctx)
```
Comment (Madanagopal19) [inline tests/pds/common.go:176]: remove tests here and below as well

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]

**Review by gauraa2 — APPROVED** [review]
LGTM

**Review by Madanagopal19 — APPROVED** [review]
LGTM

**Pull request merged**